imports/math/shared.lua

```diff
@@ -134,4 +134,46 @@ function math.clamp(val, lower, upper) -- credit https://love2d.org/forums/viewt
     return math.max(lower, math.min(upper, val))
 end
 
+---Linearly interpolates between two values over a specified duration.
+---@param start number|table|vector2|vector3|vector4  -- The starting value of the interpolation.
+---@param finish number|table|vector2|vector3|vector4 -- The ending value of the interpolation.
+---@param duration number                             -- The duration over which to interpolate.
+---@return function                                   -- a function that returns interpolated values over time.
+function math.lerp(start, finish, duration)
+	local startTime = GetGameTimer()
+	local typeStart = type(start)
+
+	if typeStart ~= type(finish) then
+		lib.print.error("start and finish must be of the same type")
+	end
+
+	local interpolate = function(s, f, t)
+		if type(s) == "number" or type(s) == "vector2" or type(s) == "vector3" or type(s) == "vector4" then
+			return s + (f - s) * t
+		elseif type(s) == "table" then
+			local result = {}
+			for i in pairs(s) do
+				result[i] = s[i] + (f[i] - s[i]) * t
+			end
+			return result
+		else
+			lib.print.error("unsupported type for lerp: " .. type(s))
+		end
```

Comment:
Can just end the `if` statement then trigger the error. Should just use a regular `error` rather than `lib.print` which is more for simple console output and doesn't have stack traces and whatnot.

I also have a typeError function defined somewhere you could copy over or expose, as this error message leaves much to be desired.

Reply:
See the type-checking note below from calling this function. Probably better to define a couple of functions in the main scope and then just assign that function to a local variable depending on the type.

Side note you have `typeStart` defined above and could reuse that.